spec/javascript/due_date.test.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import { addSeconds, addMinutes } from 'date-fns';
+import DueDate from 'hbw/components/claiming/shared/due_date';
+import Translator from 'hbw/translator';
+import { localizer } from 'hbw/init/date_localizer';
+
+describe('DueDate should render correctly', () => {
+  const env = {
+    translator: Translator.getTranslatorForLocale('en'),
+    localizer:  localizer('en')
+  };
+
+  const now = new Date();
+
+  const sec = addSeconds(now, 30);
+  const minutes = addMinutes(sec, 2);
+  const hours = addMinutes(minutes, 60 * 3);
+  const days = addMinutes(hours, 60 * 24 * 4);
+  const weeks = addMinutes(days, 60 * 24 * 7 * 5);
+  const years = addMinutes(weeks, 60 * 24 * 7 * 52 * 6);
+
+  const expired = addMinutes(now, -Math.floor(Math.random() * 60 * 24 * 7 * 52 + 1));
+
+  it('with years to deadline', () => {
+    const output = shallow(
+      <DueDate env={env} dateISO={years.toISOString()} now={now} />
+    );
+
+    expect(output.html()).toEqual('6y to deadline');
```

Comment:
I think you can move repeated code to function and make your examples body smth like
```
expect(someFuncWithShallowComponent(youDate)).toEqual('your expectation')
```